File: 19-06-2025_Partition_Array_Such_That_Maximum_Difference_Is_K.java
class Solution {
    public int partitionArray(int[] nums, int k) {
        Arrays.sort(nums);

        int i = 0;
        int j = 0;
        int res = 0;

        while (j < nums.length) {
            if ((nums[j] - nums[i]) <= k) {
                j++;
            } else {
                res++;
                i = j;
            }
        }

        return res + 1; // count the final valid group
    }
    
}

File: 19-06-2025_Partition_Array_Such_That_Maximum_Difference_is_K.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
         nums.sort()
         i = 0
         j = 0
         res = 0

         while j < len(nums):
            if nums[j] - nums[i] <= k:
                j += 1
            else:
                res += 1
                i = j

         return res + 1  # count the final valid group
